packages/app/src/systems/CRX/background/services/BackgroundService.ts

```diff
@@ -271,4 +274,33 @@ export class BackgroundService {
       url: selectedNetwork?.url,
     };
   }
+
+  async assets(_: JSONRPCParams) {
+    const assets = await AssetService.getAssets();
+
+    return assets || [];
+  }
+
+  async addAsset(
+    input: Exclude<MessageInputs['addAsset'], 'origin'>,
+    serverParams: EventOrigin
+  ) {
+    const origin = serverParams.origin;
+
+    const app = await ConnectionService.getConnection(origin);
+    if (!app) {
+      throw new Error('Unable to establish a connection. No accounts found');
```

Comment:
This is already validated by the middleware you don't need to check connection again here.